velox/core/QueryConfig.h

```diff
@@ -674,6 +677,16 @@ class QueryConfig {
     return value;
   }
 
+  int32_t sparkPartitionId() const {
+    auto id = get<int32_t>(kSparkPartitionId);
```

Comment:
Thanks for your work!
I'm not sure whether it is good to pass the id to Velox config on Gluten side then read it from config in the introduced function. cc @mbasmanova, please help review. Thanks!

Reply:
@zhli1142015, thanks so much for your information!
Yes, rand function also needs partition id to construct a new seed by combining it with user-specified seed.
I feel it may be better to maintain partition id & some other runtime metadata in an execution context (maybe need to create one), like Spark's TaskContext. Let's wait for Masha's comment.

Reply:
PR #6112 implemented rand function with a partitionIndex argument. Are partitionIndex and partition ID discussed here the same? If so, then it would be nice to find a consistent way to pass these to functions.

Reply:
Yes, they are same thing. 
Spark calls [`initialize(partitionIndex)`](https://github.com/apache/spark/blob/branch-3.3/sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala#L96C15-L96C25) to set the partition index for each function. and the partition index is retrieved from [`Partition`](https://github.com/apache/spark/blob/branch-3.3/core/src/main/scala/org/apache/spark/Partition.scala#L27) objects.
I think we would pass this information to functions by `QueryCtx`, as Gluten creates a velox task for each partition. I reuse the `QueryConfig` to pass it, as it's the only information we need to pass to fucntions. Or we can add a separate parameter for the partition index in `QueryCtx`?  @mbasmanova , could you share your thoughts on this?
Thanks.

Reply:
I think it is fine to pass partition index via QueryConfig, but then we need to start by refactoring existing rand function (in a separate PR).

Reply:
@mbasmanova, @zhli1142015, I will do some refactoring for rand function (which was implemented by me). Thanks!